web-common/src/lib/url-utils.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { getFullUrlForPath } from "@rilldata/web-common/lib/url-utils";
+import type { Page } from "@sveltejs/kit";
+import { Readable, writable } from "svelte/store";
+import { beforeAll, describe, it, SpyInstance, vi, expect } from "vitest";
+
+const pageMock: PageMock = vi.hoisted(() => ({} as any));
+
+vi.mock("$app/navigation", () => {
+  return {
+    goto: (url) => pageMock.goto(url),
+  };
+});
+vi.mock("$app/stores", () => {
+  return {
+    page: pageMock,
+  };
+});
+
+describe("url-utils", () => {
+  beforeAll(() => {
+    createPageMock();
+  });
+
+  describe("getFullUrlForPath", () => {
+    const Variations: Array<[string, string, string]> = [
+      ["/path/to/dashboard", "/new/path/dashboard", "/new/path/dashboard"],
+      ["/path/to/dashboard", "new/path/dashboard", "/new/path/dashboard"],
+      [
+        "/path/to/dashboard/?feature=all",
```

Comment:
These say `feature` but the feature flags url code says `features`, we need to pick one